### DISCUSSION TITLE
Getting 500 internal error on deployment (blocks production)

### DISCUSSION THREAD

**Discussion by Devr-pro** [sveltejs/kit#9709] [Q&A] [answered]
Hey there!, I have an app which builds and deploys (on netlify ) successfully but on visiting the page it shows 500 internal error and also logs in console that `Failed to load resource: the server responded with a status of 500 ()` about the /favicon.ico
I don't know why am I getting this error even if dont use any favicon currently 
Thanks!

**Comment by Krowemoh** [comment]
That is a strange error, you can try setting the favicon to a blank image so that the browser doesn't default to making the request.

You can add this to app.html and see if that helps.

```
<link rel="icon" href="data:;base64,=">
```

**Reply by Devr-pro** [reply]
Well, I guess favicon isnt just the alone issue causing this internal error, I really don't know where is it going wrong 
Here's the [website](https://aesthetic-cupcake-4d4274.netlify.app/)
All works perfectly in Dev environment

**Comment by ioucyf** [comment]
I have the same problem with the build step. The site works perfectly fine in dev. However, whenever I attempt to build it, it throws the error `Error: 500 /account`. The route exists, and everything there works as expected in dev without any errors or warnings. I have no idea where this error is coming from, except for the fact that it's related to prerending.

From my understanding, [according to the docs](https://kit.svelte.dev/docs/configuration#prerender), you can disable these warning with:
```js
/** @type {import('@sveltejs/kit').[Config](https://kit.svelte.dev/docs/types#public-types-config)} */
const config = {
    kit: {
        prerender: {
            // handleHttpError accepted values are:
            // 'fail' | 'warn' | 'ignore' | function handler as shown below
            handleHttpError: ({ path, referrer, message }) => {
                // ignore deliberate link to shiny 404 page
                if (path === '/not-found' && referrer === '/blog/how-we-built-our-404-page') {
                    return;
                }

                // otherwise fail the build
                throw new [Error](https://kit.svelte.dev/docs/types#app-error)(message);
            }
        }
    }
};
```

I too would very much appreciate the assistance in figuring out the cause of this issue. 🙏

**Reply by ioucyf** [reply]
Also, I tried disabling the error throwing and switched it to warn instead. The build process was able to complete, and when I ran `npm run preview` the route /account is always `404 not found` (even though the actually error code is 500 internal server error) when you try to visit it directly from the address bar. However, when you try to visit/navigate to it from the home page, it works as expected. The route still works perfectly in both cases without any issues whatsoever in dev.

**Reply by ioucyf** [reply]
I have figured out my problem.
in `/account` page, I had this import statement:

```js
// src/routes/account/+page.svelte
import { isEmail } from 'validator';
```
This works perfectly well in `npm run dev`. However, it throws an error in `npm run build`. I cannot tell whether if the issue is related to the `validatorjs` dependency or not.

So, I changed it into this:
```js
// src/routes/account/+page.svelte
import validator from 'validator';
const { isEmail } = validator;
```
This solves the problem, and `npm run build` completes as expected, with `npm run preview` showing that the app works and behaves as expected as well. 


P.S. I can see issues like this be impossibly difficult to figure out in bigger projects.

**Reply by Devr-pro** [reply]
Mine app still fails  after deployment and can't really figure out the cause, I aslo don't have any similar logic like yours which caused the issue

**Reply by ioucyf** [reply]
~~what does the build logs/error say in netlify? (assuming that's where you're deploying it)~~
nevermind, I just read what it says in the original post.

it seems to me it something similar, very inconspicuous.

my project is small, what I did was comment out the lines that had the most impact on the file and test if the error still exists, then start uncommenting in the lines one by one and that's how I was able to figure out the line (the import statement) that was causing the issue.

it's definitely not practical, but that's what I did.

**Reply by Devr-pro** [reply]
Yeah, my most of the code is just setup regarding supabase, infact I have a same type of app even bigger in size which works perfectly fine but not this!

**Comment by Devr-pro** [comment]
Here's the [repo link](https://github.com/Devr-pro/IS/tree/secondary) if anyone wants (app curently lives in secondary branch)

**Reply by ioucyf** [reply]
I have figured out the problem. The issue has to do the **_prerendering_**. From my understanding, it is required for netlify be able to serve it. However, pages with actions in sveltekit cannot be prerendered. 

So, this is what I changed with your code:

I added the following lines to the corresponding files, like so:
```ts
// src/routes/+layout.server.ts
export const prerender = true;
```
```ts
// src/routes/home/+page.server.ts
export const prerender = false;
```
```ts
// src/routes/signin/+page.server.ts
export const prerender = false;
```

Also, I'm not sure if the following would make a difference, but I added the `data-netlify="true"` attribute to all `<form>` elements [as suggested in the netlify docs for sveltekit](https://docs.netlify.com/forms/setup/#html-forms).

like so:
```html
<form method="POST" use:enhance data-netlify="true">
```

Also, for some reason, when I tried building the project on my netlify account with `pnpm run build`, it kept failing; so I changed it to `npm run build` and it started working again. If it works for you, I guess you don't have to change anything here.

with these changes, going to the root page works, and navigating to the other pages using the client-side router also works. However, if you try to directly go to the route `/signin`, netlify will fail to serve you the page, and you'll get `500 | internal error` because it's not prerendered, or at least that's what I think is happening. 

Since you're using supabase, I'd suggest not using form actions and just move the functions into +pages.svelte where the forms are and invoke the functions when the user submit as you would with vanilla js. If you did that, you'll be able to prerender these pages, and it will not be an issue navigating to any of the pages either directly or via the client-side router.

I hope this helps.

**Reply by Devr-pro** [reply]
Well, thanks for your response but my problem still persists and can't resolve it
I have same type of app with actions, using supabase and also I didn't prerender a single route, which works perfectly fine
This has something to do with my app internally or some functions causing those bugs

**Comment by Devr-pro** [comment, marked as answer]
Finally it seems that a supabase auth helper function was the issue `const {data: { session } } = await supabase.auth.getSession();`
Thanks for the replies!

**Reply by jurb** [reply]
I'm running into the same kind of errors, what change did you make to the supabase helper function to fix this if I may ask?
